perf/istio-install/setup_prometheus.sh

```diff
@@ -38,13 +38,49 @@ function install_prometheus() {
   kubectl create ns istio-prometheus || true
   kubectl label ns istio-prometheus istio-injection=disabled --overwrite
   curl -s https://raw.githubusercontent.com/coreos/prometheus-operator/v0.31.1/bundle.yaml | sed "s/namespace: default/namespace: istio-prometheus/g" | kubectl apply -f -
+
   kubectl -n istio-prometheus wait --for=condition=available --timeout=60s deploy/prometheus-operator
-  # kubectl wait is problematic, as the CRDs may not exist before the command is issued.
-  until timeout 60s kubectl get crds/prometheuses.monitoring.coreos.com; do echo "Waiting for CRDs to be created..."; done
-  until timeout 60s kubectl get crds/alertmanagers.monitoring.coreos.com; do echo "Waiting for CRDs to be created..."; done
-  until timeout 60s kubectl get crds/podmonitors.monitoring.coreos.com; do echo "Waiting for CRDs to be created..."; done
-  until timeout 60s kubectl get crds/prometheusrules.monitoring.coreos.com; do echo "Waiting for CRDs to be created..."; done
-  until timeout 60s kubectl get crds/servicemonitors.monitoring.coreos.com; do echo "Waiting for CRDs to be created..."; done
+
+  # Check CRD
+  ATTEMPTS=0
+  CMD="kubectl get crds/prometheuses.monitoring.coreos.com"
+  until $CMD || [ $ATTEMPTS -eq 60 ]; do
```

Comment:
Can you help me understand why kubectl wait doesn't work here?

Reply:
from your logs, it shows `timeout: can't execute '60s': No such file or directory`. It is trying to execute `60s` command, which is not a command.

Reply:
That's what the logs show for the previous version of the command: `until timeout 60s...`.  The comment seems to indicate this was used because `kubectl wait` was also not working, and I was wondering what the problem there had been.  It seems we're going to great lengths to effectively reproduce kubectl wait...

Reply:
I think maybe the `kubectl wait` command is not used properly. Since I remember when I use this script last week, I got `kubectl wait .... command not found` error.

Reply:
kubectl wait was added in [kubernetes 1.11](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.11.md), so unless you're using a particularly old version, you should have access to it.  I would advise using `kubectl wait` here (as in https://github.com/istio/tools/commit/d21511cab03a2f58fd3265ac9fb4ed56041bc0be#diff-ef51eaf0ba15957ed7f0b6d2f62868b9R33) unless there is a clear reason not to.

Reply:
```
$ kubectl wait crd/foobar --for=condition=established
Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "foobar" not found
```

The problem is you you can't wait on it until it exists -- at least from what I tried. Any ideas?

Reply:
Thanks for clarifying, John.  In my own tests, I see the error you describe only before the CRD apply command exits.  In other words 
```
kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/alertmanager.crd.yaml &
kubectl wait --for=condition=Established -f https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/alertmanager.crd.yaml
```
will result in the not found error, but:
```
kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/alertmanager.crd.yaml
kubectl wait --for=condition=Established -f https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/alertmanager.crd.yaml
```
will not.  Are you getting different results?  Have we seen this command result in 'Not Found' errors in practice?

Reply:
I've seen this before, yes. In the install repo we have
```
	# kubectl wait is problematic, as the pod may not exist before the command is issued.
	until timeout ${WAIT_TIMEOUT} kubectl -n ${ISTIO_CONTROL_NS} get pod/prometheus-prometheus-0; do echo "Waiting for pods to be created..."; done
	kubectl -n ${ISTIO_CONTROL_NS} wait pod/prometheus-prometheus-0 --for=condition=Ready --timeout=${WAIT_TIMEOUT}
```

If you look at the yaml we apply, https://raw.githubusercontent.com/coreos/prometheus-operator/v0.31.1/bundle.yaml, it doesn't have the CRDs. I think the operator makes these so it is async.